browser/tor/onion_location_navigation_throttle_browsertest.cc

```diff
@@ -260,6 +310,46 @@ IN_PROC_BROWSER_TEST_F(OnionLocationNavigationThrottleBrowserTest,
   EXPECT_EQ(web_contents->GetVisibleURL(), GURL(kTestOnionURL));
 }
 
+IN_PROC_BROWSER_TEST_F(OnionLocationNavigationThrottleBrowserTest,
+                       OnionLocationHeader_AutoOnionRedirect_TorWindow) {
+  browser()->profile()->GetPrefs()->SetBoolean(tor::prefs::kAutoOnionRedirect,
+                                               true);
+
+  auto* tor_browser = OpenTorWindow();
+  GURL url = test_server()->GetURL("/onion");
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(tor_browser, url));
+  EXPECT_EQ(tor_browser->tab_strip_model()->count(), 2);
+  EXPECT_EQ(tor_browser->tab_strip_model()->active_index(), 1);
+  content::WebContents* web_contents =
+      tor_browser->tab_strip_model()->GetWebContentsAt(0);
+  tor::OnionLocationTabHelper* helper =
+      tor::OnionLocationTabHelper::FromWebContents(web_contents);
+  EXPECT_FALSE(helper->should_show_icon());
+  EXPECT_TRUE(helper->onion_location().is_empty());
+
+  // Open a new tab and navigate to the url again
+  NavigateParams params(
+      NavigateParams(tor_browser, url, ui::PAGE_TRANSITION_TYPED));
+  params.disposition = WindowOpenDisposition::NEW_FOREGROUND_TAB;
+  ui_test_utils::NavigateToURL(&params);
+
+  // We stll don't close the original tab
+  // No new tab in Tor window and unique one tab per site
+  ASSERT_EQ(tor_browser->tab_strip_model()->count(), 3);
+
+  for (int i = 0; i < 3; ++i) {
```

Comment:
It looks like 3 lines of EXPECT_EQ would be clearer

Reply:
fixed and squashed in 934c12391fece233818521289babbe22382f4991